src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,138 @@
+package duke.command;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.ui.Ui;
+
+/**
+ * Commands for adding a task.
+ *
+ * @author Gao Mengqi
+ * @version CS2103T AY22/23 Semester 2
+ */
+
+public class AddCommand extends Command {
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+
+    private String taskType;
+    private String taskDesc;
+    private String deadline;
+    private String from;
+    private String by;
+
+
+    /**
+     * Constructor for Todo command.
+     *
+     * @param taskType the type of task added.
+     * @param taskDesc the description of the task.
+     */
+    public AddCommand(String taskType, String taskDesc) {
+        this.taskType = taskType;
+        this.taskDesc = taskDesc;
+    }
+
+    /**
+     * Constructor for Deadline command.
+     *
+     * @param taskType the type of task added.
+     * @param taskDesc the description of the task.
+     * @param deadline the deadline of the task.
+     */
+
+    public AddCommand(String taskType, String taskDesc, String deadline) {
+        this.taskType = taskType;
+        this.taskDesc = taskDesc;
+        this.deadline = deadline;
+    }
+
+    /**
+     * Constructor for Event command.
+     *
+     * @param taskType the type of task added.
+     * @param taskDesc the description of the task.
+     * @param from     date of the event.
+     * @param by       when the event ends.
+     */
+
+    public AddCommand(String taskType, String taskDesc, String from, String by) {
+        this.taskType = taskType;
+        this.taskDesc = taskDesc;
+        this.from = from;
+        this.by = by;
+    }
+
+
+    /**
+     * Add the task into the task list and update the task in the tasks.txt.
+     *
+     * @param tasks
+     * @param ui
+     * @param storage
+     */
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```

Comment:
Not sure I like that the method is rather long, perhaps find a way to abstract some of the execution?